packages/gatsby-source-sensenet/src/utils.ts

```diff
@@ -0,0 +1,52 @@
+import { defaultRepositoryConfiguration, ODataUrlBuilder } from '@sensenet/client-core'
+import { SourceNodesArgs } from 'gatsby'
+import fetch from 'node-fetch'
+import { PluginConfig } from './gatsby-node'
+
+export const snPrefix = 'sensenet'
+
+export const createTreeNode = async (
+  parentNode: any,
+  content: any,
+  sourceNodesArgs: Partial<SourceNodesArgs>,
+  options: PluginConfig,
+  token: string,
+) => {
+  const newNode = {
+    ...content,
+    id: sourceNodesArgs.createNodeId!(`${content.Type}-${content.Id}`),
+    internal: {
+      type: `${snPrefix}${content.Type}`,
+      contentDigest: sourceNodesArgs.createContentDigest!(content),
+      description: `${content.Type} node`,
+    },
+  }
+
+  sourceNodesArgs.actions!.createNode(newNode)
+  sourceNodesArgs.actions!.createParentChildLink({ parent: parentNode, child: newNode })
+
+  try {
+    if ((options.level && options.level > 0) || !options.level) {
```

Comment:
current level is not counted, and it would not stop at the level set in the options
what should happen when level = 0?

Reply:
After the cleanup I forgot to fix this part: I wanted to pass _level - 1_ to the function every time:
It is fixed now. If you write 0 to the config, only the children of the main path will be requested:
So if your path: '/Root/Content/SampleWorkspace/Blog' and level=0, there will be contents under the sensenetroot 
sensenetroot -> Images, but there will be no node under Images